### PR TITLE
fix assertions

### PR DESCRIPTION
**Pull request by hesmar**

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -643,8 +643,10 @@ void MainWindow::update_available_variables(PyObject *available_set)
                                    PyList_GetItem(symbol_metadata, 4)));
             request.step = PyLong_AS_LONG(
                                PyList_GetItem(symbol_metadata, 5));
-            request.pixel_layout = PyBytes_AS_STRING(
-                                       PyList_GetItem(symbol_metadata, 6));
+            PyObject *pixel_layout_bytes = PyUnicode_AsEncodedString(PyList_GetItem(symbol_metadata, 6),
```
Comment (csantosbh) [inline src/mainwindow.cpp:646]: Could you check for unicode string vs PyBytes like suggested [here](https://stackoverflow.com/a/22491037)? This should increase the robustness of the native layer to any python string format. Something along these lines:

```cpp
PyObject *pixel_layout_pyobj = PyList_GetItem(symbol_metadata, 6);
if (PyUnicode_Check(result)) {
    PyObject *pixel_layout_bytes = PyUnicode_AsEncodedString(pixel_layout_pyobj,
                                                              "ASCII",
                                                              "strict");
    request.pixel_layout = PyBytes_AS_STRING(pixel_layout_bytes);
    Py_DECREF(pixel_layout_bytes);
} else {
    assert(PyBytes_Check(pixel_layout_pyobj));
    request.pixel_layout = PyBytes_AS_STRING(pixel_layout_pyobj);
}
```

**Comment by csantosbh** [comment]
Thank you for finding and fixing this bug, @hesmar! This will probably fix the issue #11.

I have added a comment regarding your changes, can you take a look at it? If you prefer I can make the changes during the merge instead. I will also take a look for other potentially dangerous places where I need to have robustness to python unicode.

**Comment by hesmar** [comment]
I can not find your comment.

**Comment by csantosbh** [comment]
Sorry, I had forgotten to submit my review. It should be available now.

**Pull request merged**